guides/gitconfig.md

```diff
@@ -0,0 +1,36 @@
+---
+title: Managing Git Configuration
+description: Learn how to configure Git in Coder.
+---
+
+This guide will show you how to manage your Git configuration in Coder.
+
+## Personal git Configurations
+
+By default, Coder injects a .gitconfig file into each environment if there isn't
+a system-level or a user-defined .gitconfig file; the file sets the user name
```

Comment:
We need to adjust this - it's not quite accurate. We look for a global `.gitconfig` located at `~/.gitconfig`. Whether or not there's a system `/etc/gitconfig` is irrelevant.

Bare with me for a second here:

1. The build order is as follows: `git configuration`,  `coder/configure`, `personalize`. You can see this from a build log order:

![Screenshot from 2021-03-04 13-47-21](https://user-images.githubusercontent.com/11184711/110021346-38b03780-7cf0-11eb-9072-de695410cbdf.png)

2. The `git configuration` steps looks for `~/.gitconfig`
3. If no `~/.gitconfig` is found, it creates a default one using your email address and name from Coder

The line of code is literally:

```bash
if [[ -f $GLOBAL_GIT_CONFIG_PATH ]]; then
    return 0
fi
```

What this implies is we will always create one for you when you create a new environment, but it may be replaced in `coder/configure` or `personalize`, and it won't be re-injected on rebuilds if a file is already located at `~/.gitconfig`.

Reply:
I see - so rather than saying: 

>if there isn't a system-level or user-defined `.gitconfig`

it should really say:

> Coder will always inject a `.gitconfig` file in a newly created environment, but it may be overwritten in `coder/configure` or `~/personalize`, and won't be re-injected on rebuilds if a file is already located at `~/.gitconfig`.

Let me know what you think.

Reply:
Yea, something like that. How's?

> Coder will create a global git configuration file located at `~/.gitconfig` in all newly created environments.
> This step occurs before `coder/configre` (add a link) and `personalization` (add a link), which may be utilized
> to replace the default `~/.gitconfig` created by Coder. Whenever an environment is rebuilt, we'll skip creating
> a default `~/.gitconfig` if the file already exists.

What's missing is that it contains Coder email address and name - those need to be added.

CC @khorne3 for improved product copy